snforge_std/src/cheatcodes/l1_handler.cairo

```diff
@@ -5,29 +5,36 @@ use super::super::_cheatcode::handle_cheatcode;
 
 #[derive(Drop, Clone)]
 struct L1Handler {
-    contract_address: ContractAddress,
-    function_selector: felt252,
-    from_address: felt252,
-    payload: Span::<felt252>,
+    target: ContractAddress,
+    selector: felt252,
 }
 
 trait L1HandlerTrait {
-    fn new(contract_address: ContractAddress, function_selector: felt252) -> L1Handler;
-    fn execute(self: L1Handler) -> SyscallResult<()>;
+    fn new(target: ContractAddress, selector: felt252) -> L1Handler;
+    fn execute(
+        self: L1Handler, from_address: felt252, payload: Span::<felt252>
+    ) -> SyscallResult<()>;
 }
 
 impl L1HandlerImpl of L1HandlerTrait {
-    fn new(contract_address: ContractAddress, function_selector: felt252) -> L1Handler {
-        L1Handler {
-            contract_address, function_selector, from_address: 0, payload: array![].span(),
-        }
+    /// `target` - The target starknet contract address
+    /// `selector` - Selector of a `#[l1_handler]` function. Can be acquired with `selector!("function_handler_name")` marco
+    /// Returns a structure referring to a L1 handler function
+    fn new(target: ContractAddress, selector: felt252) -> L1Handler {
+        L1Handler { target, selector, }
     }
 
-    fn execute(self: L1Handler) -> SyscallResult<()> {
+    /// `self` - `L1Handler` structure referring to a L1 handler function
+    /// `from_address` - Ethereum address of the contract that you want to be the message sender
+    /// `payload` - The handlers' function arguments serialized with `Serde`
+    /// Mocks a L1 -> L2 message from Ethereum handled by the given L1 handler function
```

Comment:
I think the description on what it does should be at the top, and the bottom row under params would be the return value